src/app/parcels/batch/AddressEditCellInput.tsx

```diff
@@ -0,0 +1,214 @@
+import FreeFormTextInput from "@/components/DataInput/FreeFormTextInput";
+import { CenterComponent, GappedDiv } from "@/components/Form/formStyling";
+import { Button, Checkbox, FormControlLabel } from "@mui/material";
+import { useState } from "react";
+import { Address, BatchActionType, BatchTableDataState } from "@/app/parcels/batch/BatchTypes";
+import styled from "styled-components";
+import { ADDRESS_WIDTH } from "@/app/parcels/batch/ColumnWidths";
+import { postcodeRegex } from "@/app/clients/form/formSections/AddressCard";
+
+const GappedDivMargin = styled(GappedDiv)`
+    margin: 1rem;
+    width: calc(${ADDRESS_WIDTH}px - 2rem);
+    overflow: clip;
+`;
+
+const TopFreeFormTextInput = styled(FreeFormTextInput)`
+    margin-top: 0.5rem;
+`;
+
+interface AddressInputFields {
+    addressLine1?: string;
+    addressLine2?: string;
+    addressTown?: string;
+    addressCounty?: string;
+    addressPostcode?: string;
+}
+
+type AddressEditCellInputProps = {
+    tableState: BatchTableDataState;
+    id: number;
+    dispatchBatchTableAction: React.Dispatch<BatchActionType>;
+    simulateEscapeKeyPress: () => void;
+};
+
+const AddressEditCellInput: React.FC<AddressEditCellInputProps> = ({
+    tableState,
+    id,
+    dispatchBatchTableAction,
+    simulateEscapeKeyPress,
+}) => {
+    const defaultAddressLine1: string | undefined =
+        id !== 0
+            ? tableState.batchDataRows[id - 1].data.client.address?.addressLine1 ?? undefined
+            : tableState.overrideDataRow.data.client.address?.addressLine1 ?? undefined;
```

Comment:
Possibly could extract `tableState.batchDataRows[id - 1].data.client.address?` into its own variable, just to not keep writing the same thing? Same for override data row.